protocol/deploys/README.txt

```diff
@@ -0,0 +1,9 @@
+This directory contains a complete list of all on chain contract deployments
+
+Directory Structure
+  - `mainnet/` all contracts deployed on mainnet
+  - `goreli/` all contracts deployed on goreli testnet
+  - `contract-address.ts` all active deployed contracts (NOTE: to be deleted/unified with dapp)
+  - `helpers.ts` common code to help deployments
+  - `[0-9]+-some-script-name.ts` last migration script run. We try and make the script generic s.t it will run on both
```

Comment:
nit: Should make this file a markdown

In stax the testnet scripts diverge from mainnet, because there's a bunch of other things we need to do in testnet when the upstream protocol isn't deployed there.

So will probably still need a specific set of `goerli/[0-9]+-some-script-name.ts` scripts. A chunk of them will be the same though.

Similarly with `contract-addresses.ts`, because there are a bunch of addresses specific only to the testnet setup, I'm leaning on creating a testnet specific interface rather than a bunch of empty string/optionals. Can see the gmx branch for a WIP setup of that

Reply:
Also, if we have changes to a contract over time - eg a v2 which adds extra params to the constructor -- are you thinking that there should be another *new* deploy script so we can keep a fully reproducible history? Or just relying on git history for that?

I guess LEGACY-DEPLOYED.md should capture the v1 contract too

Reply:
I agree to the above :)